src/resolutionError.ts

```diff
@@ -22,6 +22,10 @@ const HandlersByCode = {
     `Domain ${params.domain} has no ${params.currencyTicker} attached to it`,
   NamingServiceDown: (params: { method: string }) =>
     `${params.method} naming service is down at the moment`,
+  UnsupportedCurrency: (params: { currencyTicker: string }) =>
+    `${params.currencyTicker} is not supported`,
+  IncorrectResolverInterface: (params: { method: string }) =>
```

Comment:
Please throw this type of error for zns when resolver returns `undefined` for `records.

Reply:
added a test to check for IncorrectResolverInterface on zns side. It passes by itself correctly, but when run with other tests breaks randomly with wrong code error. Not sure what can cause the issue. I left it under comments. I will merge this and resolve it later.